From eb1819537c967acd3f1c1e57007d5cb306c74a03 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:05:58 -0400
Subject: [usb] core: Move USB_MAXENDPOINTS definitions to usb.h

Message-id: <1410887463-20674-287-git-send-email-dzickus@redhat.com>
Patchwork-id: 95651
O-Subject: [RHEL7 PATCH 286/591] usb-core: Move USB_MAXENDPOINTS definitions to usb.h
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 8f5d35441ff26b31e3812556ce468c76f1eb216b
Author: Hans de Goede <hdegoede@redhat.com>
Date:   Wed Oct 9 17:19:24 2013 +0200

    usb-core: Move USB_MAXENDPOINTS definitions to usb.h

    So that it can be used in other places too.

    Signed-off-by: Hans de Goede <hdegoede@redhat.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/core/config.c b/drivers/usb/core/config.c
index 8d72f0c..14ba398 100644
--- a/drivers/usb/core/config.c
+++ b/drivers/usb/core/config.c
@@ -10,7 +10,6 @@
 
 
 #define USB_MAXALTSETTING  128 /* Hard limit */
-#define USB_MAXENDPOINTS  30 /* Hard limit */
 
 #define USB_MAXCONFIG   8 /* Arbitrary limit */
 
diff --git a/include/linux/usb.h b/include/linux/usb.h
index 7454865..4c53d54 100644
--- a/include/linux/usb.h
+++ b/include/linux/usb.h
@@ -202,6 +202,8 @@ static inline void usb_set_intfdata(struct usb_interface *intf, void *data)
 struct usb_interface *usb_get_intf(struct usb_interface *intf);
 void usb_put_intf(struct usb_interface *intf);
 
+/* Hard limit */
+#define USB_MAXENDPOINTS 30
 /* this maximum is arbitrary */
 #define USB_MAXINTERFACES 32
 #define USB_MAXIADS  (USB_MAXINTERFACES/2)
-- 
1.7.1